corehq/apps/app_manager/views/utils.py

```diff
@@ -373,18 +373,38 @@ def update_linked_app(app, user_id):
                 ).format(ucr_id=str(e))
             )
 
-    if app.master_is_remote:
-        try:
-            pull_missing_multimedia_for_app(app)
-        except RemoteRequestError:
-            raise AppLinkError(_(
-                'Error fetching multimedia from remote server. Please try again later.'
-            ))
+        if app.master_is_remote:
+            try:
+                pull_missing_multimedia_for_app(app, old_multimedia_ids)
+            except RemoteRequestError:
+                raise AppLinkError(_(
+                    'Error fetching multimedia from remote server. Please try again later.'
+                ))
+
+        # reapply linked application specific data
+        app.reapply_overrides()
 
     app.domain_link.update_last_pull('app', user_id, model_details=AppLinkDetail(app_id=app._id))
 
-    # reapply linked application specific data
-    app.reapply_overrides()
+
+def pull_missing_multimedia_for_app_and_notify(domain, app_id, email):
```

Comment:
Same as https://github.com/dimagi/commcare-hq/pull/25010/files#r310487348, can this be moved to a `views/utils` under the `linked_domain` app or `linked_apps` app?

Reply:
sounds good to me

Reply:
https://github.com/dimagi/commcare-hq/pull/25010/commits/1ce375dc4af17100c6d94109ed366d2aacaa0743